a_test.go

```diff
@@ -183,42 +186,360 @@ func TestA_uses_Logger_dynamic_interface(t *testing.T) {
 	}
 }
 
+func TestA_WithContext(t *testing.T) {
+	type ctxKeyT struct{}
+	var ctxKey ctxKeyT
+
+	cases := []struct {
+		name       string
+		parallel   bool
+		throwPanic bool
+	}{
+		{
+			name:       "parallel and panic",
+			parallel:   true,
+			throwPanic: true,
+		},
+		{
+			name:     "parallel",
+			parallel: true,
+		},
+		{
+			name: "sequential",
+		},
+		{
+			name:       "sequential and panic",
+			throwPanic: true,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			flow, task, childTasksResults, _ := prepareFlowAndTask(t, 3, ctxKey, func(*goyek.A, int) {})
+			childTasks := prepareTasks(t, flow, len(childTasksResults), c.parallel, task, func(i int) func(a *goyek.A) {
+				return func(a *goyek.A) {
+					a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+
+					if c.throwPanic {
+						a.Cleanup(func() {
+							panic("some panic")
+						})
+					}
+
+					prevCtx := a.Context()
+					newA := a.WithContext(context.WithValue(prevCtx, ctxKey, i))
+					assertEqual(t, newA.Name(), a.Name(), "name changed for "+a.Name())
+					assertEqual(t, a.Context(), prevCtx, "context changed for "+a.Name())
+
+					task.Action()(newA)
+				}
+			})
+
+			main := flow.Define(goyek.Task{
+				Name: "main",
+				Deps: childTasks,
+			})
+
+			chkFunc := assertPass
+			if c.throwPanic {
+				chkFunc = assertFail
+			}
+			chkFunc(t, flow.Execute(context.Background(), []string{main.Name()}), "flow execute")
+
+			for index, result := range childTasksResults {
+				chkFunc := assertTrue
+				if !c.parallel && c.throwPanic && index != 0 { // only first will be true if not parallel and with panic
+					chkFunc = assertFalse
+				}
+
+				chkFunc(t, result, "child result "+strconv.Itoa(index)+" is incorrect")
+			}
+		})
+	}
+}
+
+func TestA_WithContextFatal(t *testing.T) {
+	type ctxKeyT struct{}
+	var ctxKey ctxKeyT
+
+	cases := []struct {
+		name     string
+		parallel bool
+	}{
+		{
+			name:     "parallel",
+			parallel: true,
+		},
+		{
+			name: "sequential",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			flow, task, childTasksResults, loggerSpy := prepareFlowAndTask(t, 3, ctxKey, func(a *goyek.A, i int) {
+				a.Fatal(i)
+
+				a.Cleanup(func() {
+					failed := a.Failed()
+					assertTrue(t, failed, "a.Failed() should return true for "+a.Name())
+				})
```

Comment:
This will never be called. You need to register cleanup before. `a.Fatal`.

Reply:
Skip also